### PR TITLE
[test] Don't assume vSphere network name

### PR DESCRIPTION
**Pull request by aravindhp**
Instead get it from the env var $LEASED_RESOURCE for CI and assume "dev-segment" otherwise.

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
LGTM

File: test/e2e/providers/vsphere/vsphere.go [hunk 1]
```diff
@@ -63,6 +65,21 @@ func newVSphereMachineProviderSpec(clusterID string) (*vsphere.VSphereMachinePro
 	}, nil
 }
 
+// getNetwork returns the network that needs to be used in the MachineSet
+func getNetwork() string {
+	// Default network for dev environment
+	networkSegment := "dev-segment"
+	if os.Getenv("OPENSHIFT_CI") == "true" {
+		// $LEASED_RESOURCE holds the network the CI cluster is in
+		networkSegment = os.Getenv("LEASED_RESOURCE")
```
Comment (sebsoto) [inline test/e2e/providers/vsphere/vsphere.go:74]: Is there documentation that can be linked to regarding this?

**Review by sebsoto — APPROVED** [review]

File: test/e2e/providers/vsphere/vsphere.go (hunk 1 above)
Comment (aravindhp) [reply test/e2e/providers/vsphere/vsphere.go:74]: The only references I found were [quota-and-leases/#directions-for-job-authors](https://docs.ci.openshift.org/docs/architecture/quota-and-leases/#directions-for-job-authors) and [step-registry#leases](https://docs.ci.openshift.org/docs/architecture/step-registry/#leases). Both did not have anything vSphere specific and linking to either of these does not make much sense. @jcpowermac might know of some other docs.

**Pull request merged**